frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.unit.spec.tsx

```diff
@@ -0,0 +1,535 @@
+      expect(onChange).toHaveBeenLastCalledWith(["abc"]);
+      expect(onBlur).toHaveBeenCalled();
+    });
+
+    it("should allow to add multiple values", async () => {
+      const { onChange, onFocus, onBlur } = await setupStringPicker({
+        query,
+        stageIndex,
+        column,
+        values: ["abc"],
+      });
+
+      userEvent.type(screen.getByLabelText("Filter value"), "bce");
+      userEvent.tab();
+
+      expect(onFocus).toHaveBeenCalled();
+      expect(onChange).toHaveBeenLastCalledWith(["abc", "bce"]);
+      expect(onBlur).toHaveBeenCalled();
+    });
+
+    it("should not allow to add empty values", async () => {
+      const { onChange, onFocus, onBlur } = await setupStringPicker({
+        query,
+        stageIndex,
+        column,
+        values: [],
+      });
+
+      const input = screen.getByPlaceholderText("Enter some text");
+      userEvent.type(input, "abc");
+      userEvent.clear(input);
+      userEvent.tab();
+
+      expect(onFocus).toHaveBeenCalled();
+      expect(onChange).toHaveBeenLastCalledWith([]);
+      expect(onBlur).toHaveBeenCalled();
+    });
+
+    it("should allow to remove a value", async () => {
+      const { onChange } = await setupStringPicker({
+        query,
+        stageIndex,
+        column,
+        values: ["abc", "bce"],
+      });
+
+      userEvent.type(screen.getByLabelText("Filter value"), "{backspace}");
+
+      expect(onChange).toHaveBeenLastCalledWith(["abc"]);
+    });
+  });
+});
+
+describe("NumberFilterValuePicker", () => {
+  const query = createQuery();
+  const stageIndex = 0;
+  const availableColumns = Lib.filterableColumns(query, stageIndex);
+  const findColumn = columnFinder(query, availableColumns);
+
+  beforeAll(() => {
+    jest.useFakeTimers({ advanceTimers: true });
+  });
+
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+
+  describe("list values", () => {
+    const column = findColumn("ORDERS", "QUANTITY");
+
+    it("should allow to pick a list value", async () => {
+      const { onChange } = await setupNumberPicker({
+        query,
+        stageIndex,
+        column,
+        values: [],
+        fieldValues: createMockFieldValues({
+          field_id: ORDERS.QUANTITY,
+          values: [[10], [20], [30]],
+        }),
+      });
+
+      userEvent.click(screen.getByText("20"));
+
+      expect(onChange).toHaveBeenCalledWith([20]);
+    });
+
+    it("should handle field values remapping", async () => {
+      const { onChange } = await setupNumberPicker({
+        query,
+        stageIndex,
+        column,
+        values: [10],
+        fieldValues: createMockFieldValues({
+          field_id: ORDERS.QUANTITY,
+          values: [
+            [10, "To-do"],
+            [20, "In-progress"],
+            [30, "Completed"],
+          ],
+        }),
+      });
+      expect(screen.getByRole("checkbox", { name: "To-do" })).toBeChecked();
+      expect(screen.getByRole("checkbox", { name: "To-do" })).toBeChecked();
```

Comment:
Did you mean the second line to assert that "In-progress" is not checked?

Reply:
Fixed